agreementbot/api.go

```diff
@@ -0,0 +1,242 @@
+package agreementbot
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/gorilla/mux"
+	"github.com/boltdb/bolt"
+	"github.com/golang/glog"
+	"github.com/open-horizon/anax/citizenscientist"
+	"github.com/open-horizon/anax/config"
+	"github.com/open-horizon/anax/events"
+	"github.com/open-horizon/anax/worker"
+	"net/http"
+	"sort"
+)
+
+type API struct {
+	worker.Manager // embedded field
+	db             *bolt.DB
+}
+
+func NewAPIListener(config *config.HorizonConfig, db *bolt.DB) *API {
+	messages := make(chan events.Message)
+
+	listener := &API{
+		Manager: worker.Manager{
+			Config:   config,
+			Messages: messages,
+		},
+
+		db: db,
+	}
+
+	listener.listen(config.AgreementBot.APIListen)
+	return listener
+}
+
+// Worker framework functions
+func (a *API) Messages() chan events.Message {
+	return a.Manager.Messages
+}
+
+func (a *API) NewEvent(ev events.Message) {
+	if a.Config.AgreementBot.APIListen == "" {
+		return
+	}
+
+	return
+}
+
+func (a *API) listen(apiListen string) {
+	glog.Info("Starting AgreementBot API server")
+
+	// If there is no Agbot config, we will terminate
+	if a.Config.AgreementBot.APIListen == "" {
```

Comment:
oh yeah, I didnt even notice that it was passed into the function. I will fix it in my next PR.